frontend/rolecall/src/app/cast/cast-drag-and-drop.component.ts

```diff
@@ -127,206 +153,150 @@ export class CastDragAndDrop implements OnInit {
       uuid: this.selectedCastUUID,
       segment: this.cast.segment,
       name: this.cast.name,
-      filled_positions: this.data.map((val, posInd) => {
-        let groupArr = Array(this.getMaxNumberInDancerPositionForData(posInd)).fill({ group_index: 0, members: [] });
-        groupArr = groupArr.map((val, ind) => {
+      filled_positions: this.castPositions.map((uiPos: UICastPosition, uiPosIx: number) => {
+        let subCasts: CastSubCast[] = new Array(CAST_COUNT).fill([]);
+        subCasts = subCasts.map((subCast, subCastIx) => {
           return {
-            group_index: ind,
+            group_index: subCastIx,
             members: []
           };
         });
-        this.data[posInd].forEach((val, dancerPosIndex) => {
-          val.forEach((val2, subcastInd) => {
-            groupArr[subcastInd].members.push(
-              {
-                uuid: val2.uuid,
-                position_number: dancerPosIndex
-              }
-            )
-          });
-        });
+        for (let subCastIx = 0; subCastIx < subCasts.length; subCastIx++) {
+          for (let dancerIx = 0; dancerIx < uiPos.castSize; dancerIx++) {
+            const dancer = uiPos.castRows[dancerIx].subCastDancers[subCastIx];
+            if (dancer !== undefined) {
```

Comment:
Looks like no harm in simplifying to `if (dancer) {` here?

Reply:
Done.